processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -70,6 +70,7 @@
   public static final String BY_SEGMENT_KEY = "bySegment";
   public static final String BROKER_SERVICE_NAME = "brokerService";
   public static final String IN_SUB_QUERY_THRESHOLD_KEY = "inSubQueryThreshold";
+  public static final String PLAN_TIME_BOUNDARY_SQL = "planTimeBoundarySql";
```

Comment:
can you add documentation for this context? Also can this be renamed to 
`enableTimeBoundaryPlanning` or `enableTimeBoundaryConversion` ? I am suggesting it since other boolean flags follow similar convention. Also `sql` word in there is not needed.

Reply:
I had added `sql` to point that this config is only applicable to sql planning - the native query runs are independent of this flag.
But seems right as you said that no other property also has this distinction - which upon rereading of the query context flags made me realise that there's no obvious way to in the name to know that whether a property is applicable only to SQL or both.
There's `PlannerConfig` which has a `druid.sql.planner` prefix for all properties but that doesn't allow a direct query level override.
Anyways, I will rename to `enableTimeBoundaryPlanning` to adhere to the current convention - thanks for the suggestion